mirrord/kube/src/api/kubernetes.rs

```diff
@@ -331,30 +334,43 @@ pub async fn create_kube_config<P>(
     kube_context: Option<String>,
 ) -> Result<Config>
 where
-    P: AsRef<str>,
+    P: AsRef<OsStr>,
 {
     let kube_config_opts = KubeConfigOptions {
         context: kube_context,
         ..Default::default()
     };
 
-    let mut config = match kubeconfig {
-        Some(kubeconfig) => {
-            let kubeconfig = shellexpand::full(&kubeconfig)
-                .map_err(|e| KubeApiError::ConfigPathExpansionError(e.to_string()))?;
-            let parsed_kube_config = Kubeconfig::read_from(kubeconfig.deref())?;
-            Config::from_custom_kubeconfig(parsed_kube_config, &kube_config_opts).await?
-        }
-        _ => {
-            if kube_config_opts.context.is_some() {
-                // if context is set, it's not in cluster so it has to be a kubeconfig.
-                Config::from_kubeconfig(&kube_config_opts).await?
-            } else {
-                // if context isn't set and user doesn't specify a kubeconfig, we infer which tries
-                // local kube or incluster configuration.
-                Config::infer().await?
-            }
-        }
+    // parse kubeconfig the same way as KUBECONFIG is parsed by `kube-client`, supporting
+    // colon-separated lists of paths. Borrowed affectionately & with love from
+    // https://docs.rs/kube/latest/kube/config/struct.Kubeconfig.html#method.from_env
+    let mut config = if let Some(kubeconfig) = kubeconfig
+        && let paths = std::env::split_paths(&kubeconfig)
+            .filter_map(|p| {
+                let string = p.as_os_str().to_string_lossy().into_owned();
+                string.is_empty().not().then(|| string)
```

Comment:
```suggestion
                string.is_empty().not().then_some(string)
```

`then_some` if you're not putting the conversion function inside `then` I guess.

I would suggest the name `path_str` instead of `string`.